src/query/src/range_select/plan.rs

```diff
@@ -1339,29 +1400,76 @@ mod test {
             ScalarValue::UInt8(None),
             ScalarValue::UInt8(Some(9)),
         ];
-        Fill::Null.apply_fill_strategy(&mut test1).unwrap();
+        Fill::Null.apply_fill_strategy(&[], &mut test1).unwrap();
         assert_eq!(test1[1], ScalarValue::UInt8(None));
-        Fill::Prev.apply_fill_strategy(&mut test1).unwrap();
+        Fill::Prev.apply_fill_strategy(&[], &mut test1).unwrap();
         assert_eq!(test1[1], ScalarValue::UInt8(Some(8)));
         test1[1] = ScalarValue::UInt8(None);
         Fill::Const(ScalarValue::UInt8(Some(10)))
-            .apply_fill_strategy(&mut test1)
+            .apply_fill_strategy(&[], &mut test1)
             .unwrap();
         assert_eq!(test1[1], ScalarValue::UInt8(Some(10)));
-        test1[1] = ScalarValue::UInt8(None);
-        assert_eq!(
-            Fill::Linear
-                .apply_fill_strategy(&mut test1)
-                .unwrap_err()
-                .to_string(),
-            "Execution error: RangePlan: Apply Fill LINEAR strategy on Non-floating type"
-        );
-        let mut test2 = vec![
-            ScalarValue::Float32(Some(8.0)),
+    }
+
+    #[test]
+    fn test_fill_linear() {
```

Comment:
Can this handle changing slope? E.g., `1, x, 3, x, 9`

Reply:
Test has been updated

Reply:
I think you should generate interpolated data with different slopes, not overwriting existing points. This is what InfluxQL does:

<img width="553" alt="image" src="https://github.com/GreptimeTeam/greptimedb/assets/15380403/bb278041-9e30-407f-a231-0af7eb19c628">